### ISSUE TITLE
IDE0306 code fixer produces invalid code (CS9215)

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#77174]
**Version Used**: Compiler version: '4.13.0-3.25105.4 (400efa2d)'. Language version: 13.0.

**Steps to Reproduce**:

1. Create a new .NET 9 console project
2. Paste this code:

```csharp
_ = Test(new Dictionary<string, object>());

static Dictionary<string, object> Test(IReadOnlyDictionary<string, object> d)
{
    return new(d);
}
```

A minimal repro, with source-code provided, is ideal.  Most compiler/language issues can be distilled into a snippet that can be pasted into [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRhQG4BYAKHIH0cBeHAFRgGcAXACgDsYB3HAEQCWYFgIgcAhlACeAHgJYANDgjAAVjGEA+NgEodpCmQJJ+QkWMmz5Sleq0Nm7AJIAlGOIAmAeQ4AbKYLCohLScviKymoaLJo4HjrkAN7kOKl4AOw4XNxs8QYAvkA===).

**Diagnostic Id**: IDE0306, "Collection initialization can be simplified"

**Expected Behavior**:

I shouldn't be getting this warning here.

**Actual Behavior**:

I get the IDE0306 warning:

![Image](https://github.com/user-attachments/assets/9c581648-3da9-46c6-9de9-bae74b66533b)

Applying the code fixer produces invalid code:

![Image](https://github.com/user-attachments/assets/b1c9f03b-be2f-44de-9be2-fe592926d6d0)

cc. @CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
I don't repro this on 17.14.9p1

<img width="1566" height="568" alt="Image" src="https://github.com/user-attachments/assets/3e85c4b8-2ae2-45a4-abab-ee2c35da7180" />
